### DISCUSSION TITLE
[FEATURE REQUEST]: Recognize `pyarrow` dtype in `pandas`

### DISCUSSION THREAD

**Discussion by csubhodeep** [duckdb/duckdb#7399] [Q&A] [answered]
Currently for the following snippet from the official documentations
```python
my_df = pd.DataFrame.from_dict({'a': [42]}, dtype="float32[pyarrow]")
result = duckdb.sql("SELECT * FROM my_df")
```
I get the following exception
```shell
NotImplementedException: Not implemented Error: Data type 'float[pyarrow]' not recognized
```
Since `pandas>=2.0.0` most dtypes can be potentially backed by [`pyarrow`](https://pandas.pydata.org/docs/user_guide/pyarrow.html) it could be a common place to use it. So would be really nice to have it interpretable in `duckdb` as well. 

Also ideally, the datatypes should be mapped back correctly once the `.df()` method is called on the `duckdb.DuckDBPyRelation` object (e.g. the `result` variable in the above snippet). Thanks !

**Comment by Tishj** [comment, marked as answer]
https://github.com/duckdb/duckdb/pull/7005 👍

**Reply by Tishj** [reply]
They are not mapped back correctly however, once the DataFrame gets scanned by us we lose all context of the origins, `.df` outputs a numpy-backed dataframe.
I recommend outputting arrow and constructing a pyarrow backed pandas dataframe from that for now.
